libkbfs/block_tree_test.go

```diff
@@ -0,0 +1,65 @@
+package libkbfs
+
+import (
+	"github.com/golang/mock/gomock"
+	"reflect"
+	"testing"
+)
+
+func makePaths(childIndices [][]int,
+	mockBlock BlockWithPtrs) [][]parentBlockAndChildIndex {
+	result := [][]parentBlockAndChildIndex(nil)
+	for _, indexList := range childIndices {
+		path := []parentBlockAndChildIndex(nil)
+		for _, childIndex := range indexList {
+			path = append(path, parentBlockAndChildIndex{
+				pblock:     mockBlock,
+				childIndex: childIndex,
+			})
+		}
+		result = append(result, path)
+	}
+	return result
+}
+
+func TestCheckForHolesAndTruncate(t *testing.T) {
+	// Make a mock block that believes itself to have three children.
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockBlock := NewMockBlockWithPtrs(mockCtrl)
+	mockBlock.EXPECT().NumIndirectPtrs().Return(4).MinTimes(1)
+
+	goodExamples := [][][]int{
+		{{3, 2}, {3, 3}},
+		{{2, 3}, {3, 0}},
+		{{1, 0}, {1, 1}, {1, 2}, {1, 3}, {2, 0}, {2, 1}},
+		{{1, 2, 3}, {1, 3, 0}, {1, 3, 1}, {1, 3, 2}, {1, 3, 3}, {2, 0, 0}},
+	}
+	badExamples := [][][]int{
```

Comment:
Slight preference to moving `badExamples` and `correctLengths` definitions to right above the bad examples loop.